src/Shared/E2ETesting/E2ETesting.targets

```diff
@@ -0,0 +1,79 @@
+<Project>
+  <!-- Version of this SDK is set in global.json -->
+  <Sdk Name="Yarn.MSBuild" />
+
+  <!-- Ensuring that everything is ready before build -->
+
+  <Target Name="EnsureNodeJSRestored" BeforeTargets="Build">
+    <Message Text="Running yarn install on $(MSBuildProjectFile)" Importance="High" />
+    <Yarn Command="install" />
+  </Target>
+
+  <!-- Running prerequisites -->
+
+  <Target Name="EnsurePrerequisites" Condition="'$(EnforcePrerequisites)' == 'True'" BeforeTargets="EnsureNodeJSRestored">
+    <PropertyGroup>
+      <_PackageJson>$(MSBuildProjectDirectory)\package.json</_PackageJson>
+    </PropertyGroup>
+
+    <!-- JAVA -->
+    <Message Importance="High" Text="Ensuring JAVA is available" />
+    <Exec Command="java -version" />
```

Comment:
Can we capture the exit code and print a message that says what needs to be done resolve this?

Reply:
Why not `<Exec Command="${JAVA_HOME}/bin/java.exe -version"/>` ?

Reply:
Because this needs to run cross-platform and needs to be on the PATH for https://github.com/aspnet/AspNetCore/pull/8188/files?utf8=%E2%9C%93&diff=unified#diff-4035093707e8bfd6d9ba2a44912dce70R9 to work